Update cache when file size === 0

artonge opened the pull request:
The conditions were false when `$result === 0`.
`$results` here contains the number of written bits.
The correct way of checking for operation success is to check if `$result === false`

## Question

- Not sure what `$result` is supposed to contain for other operation, but it is probably safe to assume that none were returning 0 for saying error

## How to test

1. Create a new plain text file toto.txt
2. Add a dummy content and wait for it to be saved
3. Clear the content and wait for it to be savec
4. Refresh view and constat that file size and mtime is not up to date.

artonge commented:
/backport to stable25

artonge commented:
/backport to stable24

szaimen reviewed (approved):
LGTM but didnt test

PVince81 reviewed (approved):
:+1:

PVince81 commented:
great find!

PVince81 commented:
/backport to stable23

PVince81 commented:
backporting further because file cache inconsistencies can be nasty with some versions of the desktop client

PVince81 commented:
@artonge linter not happy though, maybe remove the unrelated formatting changes ?

artonge committed:
Update cache when file size === 0

The conditions were false when $result === 0.
$results here contains the number of written bits.
The correct way of checking for operation success is to check if $result === false

Signed-off-by: Louis Chemineau <louis@chmn.me>
